libs/langchain-google-cloud-sql-pg/src/tests/engine.test.ts

```diff
@@ -82,5 +89,140 @@ describe("PostgresEngine Instance creation", () => {
     } catch (error) {
       throw new Error(`Error on closing connection: ${error}`);
     }
-  }) 
-})
\ No newline at end of file
+  }); 
+
+  test('should create a PostgresEngine Instance through from_engine method', async () => {
+    PostgresEngine.connector = new Connector();
+    const clientOpts = await PostgresEngine.connector.getOptions({
+      instanceConnectionName: `${process.env.PROJECT_ID}:${process.env.REGION}:${process.env.INSTANCE_NAME}`,
+      ipType: IpAddressTypes.PUBLIC,
+      authType: AuthTypes.PASSWORD
+    })
+
+    const dbConfig: knex.Knex.Config<any> = {
+      client: 'pg',
+      connection: {
+        ...clientOpts,
+        password: process.env.PASSWORD,
+        user: process.env.DB_USER,
+        database: process.env.DB_NAME,
+      },
+    };
+
+    const engine = knex(dbConfig)
+    PEInstance = await PostgresEngine.from_engine(engine)
+
+    const {rows} = await PEInstance.testConnection();
+    const currentTimestamp = rows[0].currenttimestamp;
+    expect(currentTimestamp).toBeDefined();
+
+    try {
+      await PEInstance.closeConnection();
+    } catch (error) {
+      throw new Error(`Error on closing connection: ${error}`);
+    }
+  });
+
+  test('should throw an error if the URL passed to from_engine_args does not have the driver', async () => {
+    const url = "";
+
+    async function createInstance() {
+      PEInstance = await PostgresEngine.from_engine_args(url);
+    }
+
+    await expect(createInstance).rejects.toBe("Driver must be type 'postgresql+asyncpg'");
+  });
+
+  test('should create a PostgresEngine Instance through from_engine_args using a URL', async () => {
+    const url = process.env.PG_URL ?? "";
```

Comment:
Can you construct this like "postgresql+asyncpg://{USER}:{PASSWORD}@{HOST}:5432/{DATABASE_NAME}", so the only new env var is the Host?